### ISSUE TITLE
Code generated from npx oclif generate cannot be packed

### ISSUE THREAD

**Issue by drewjenkins** [oclif/oclif#1161]
Do you want to request a *feature* or report a *bug*?
-----------------------------------------------------
Bug

What is the current behavior?
-----------------------------
Trying to pack the project generated from npx oclif generate creates an error
1. `npx oclif generate` and follow the prompts, then cd into the created folder
2. `npx oclif pack tarballs` (Also errors on other pack types)

ERROR
```
(node:59893) SyntaxError Plugin: node: Cannot use import statement outside a module
module: @oclif/core@2.9.4
task: toCached
plugin: node
root: /my/root/node
See more details with DEBUG=*
(Use `node --trace-warnings ...` to show where the warning was created)
(node:59893) SyntaxError Plugin: node: Cannot use import statement outside a module
module: @oclif/core@2.9.4
task: toCached
plugin: node
root: my/root/node
See more details with DEBUG=*
    Error: Command failed: git rev-parse --short HEAD
    fatal: not a git repository (or any of the parent directories): .git

    Code: 128
```

What is the expected behavior?
------------------------------
Expect a tarball to be created with no error.

Version Info
oclif/3.10.0 darwin-arm64 node-v16.17.1

**Comment by drewjenkins** [comment]
@mdonnalley Here's a video 


https://github.com/oclif/oclif/assets/5521743/0279e43e-403d-4fca-ac0d-f6905b94bc1b

**Comment by drewjenkins** [comment]
It looks as though when generating the tarball, it is trying to include the typescript files instead of the build /dist files maybe?

```shell
andrewjenkins@MacBook-Pro node % DEBUG=* oclif pack tarballs                                              
  config reading core plugin /MyPath/.nvm/versions/node/v18.17.0/lib/node_modules/oclif +0ms
  config loadJSON /MyPath/.nvm/versions/node/v18.17.0/lib/node_modules/oclif/package.json +0ms
  config loadJSON /MyPath/.nvm/versions/node/v18.17.0/lib/node_modules/oclif/oclif.manifest.json +0ms
  config loadJSON /MyPath/.nvm/versions/node/v18.17.0/lib/node_modules/oclif/.oclif.manifest.json +0ms
  config:oclif using manifest from /MyPath/.nvm/versions/node/v18.17.0/lib/node_modules/oclif/.oclif.manifest.json +0ms
  config reading user plugins pjson /MyPath/.local/share/oclif/package.json +0ms
  config loadJSON /MyPath/.local/share/oclif/package.json +0ms
  config loading plugins [
  '@oclif/plugin-help',
  '@oclif/plugin-warn-if-update-available',
  '@oclif/plugin-not-found'
] +2ms
  config reading core plugin /MyPath/.nvm/versions/node/v18.17.0/lib/node_modules/oclif/node_modules/@oclif/plugin-help +0ms
  config loadJSON /MyPath/.nvm/versions/node/v18.17.0/lib/node_modules/oclif/node_modules/@oclif/plugin-help/package.json +0ms
  config reading core plugin /MyPath/.nvm/versions/node/v18.17.0/lib/node_modules/oclif/node_modules/@oclif/plugin-warn-if-update-available +0ms
  config loadJSON /MyPath/.nvm/versions/node/v18.17.0/lib/node_modules/oclif/node_modules/@oclif/plugin-warn-if-update-available/package.json +0ms
  config reading core plugin /MyPath/.nvm/versions/node/v18.17.0/lib/node_modules/oclif/node_modules/@oclif/plugin-not-found +0ms
  config loadJSON /MyPath/.nvm/versions/node/v18.17.0/lib/node_modules/oclif/node_modules/@oclif/plugin-not-found/package.json +0ms
  config loadJSON /MyPath/.nvm/versions/node/v18.17.0/lib/node_modules/oclif/node_modules/@oclif/plugin-help/oclif.manifest.json +0ms
  config loadJSON /MyPath/.nvm/versions/node/v18.17.0/lib/node_modules/oclif/node_modules/@oclif/plugin-not-found/oclif.manifest.json +0ms
  config loadJSON /MyPath/.nvm/versions/node/v18.17.0/lib/node_modules/oclif/node_modules/@oclif/plugin-warn-if-update-available/oclif.manifest.json +0ms
  config:@oclif/plugin-not-found using manifest from /MyPath/.nvm/versions/node/v18.17.0/lib/node_modules/oclif/node_modules/@oclif/plugin-not-found/oclif.manifest.json +0ms
  config:@oclif/plugin-help using manifest from /MyPath/.nvm/versions/node/v18.17.0/lib/node_modules/oclif/node_modules/@oclif/plugin-help/oclif.manifest.json +0ms
  config:@oclif/plugin-warn-if-update-available using manifest from /MyPath/.nvm/versions/node/v18.17.0/lib/node_modules/oclif/node_modules/@oclif/plugin-warn-if-update-available/oclif.manifest.json +0ms
  config config done +3ms
  config start init hook +0ms
  oclif:@oclif/plugin-warn-if-update-available:hooks:init start (require) /MyPath/.nvm/versions/node/v18.17.0/lib/node_modules/oclif/node_modules/@oclif/plugin-warn-if-update-available/lib/hooks/init/check-update.js +0ms
  oclif:@oclif/plugin-warn-if-update-available:hooks:init done +1ms
  config init hook done +107ms
  config runCommand pack:tarballs [] +0ms
  config:oclif (require) /MyPath/.nvm/versions/node/v18.17.0/lib/node_modules/oclif/lib/commands/pack/tarballs.js +142ms
  config start prerun hook +24ms
  config prerun hook done +0ms
  oclif:pack:tarballs init version: @oclif/core@2.11.3 argv: [] +0ms
  config reading core plugin /MyPath/src/path/pluto-cli/node +0ms
  config loadJSON /MyPath/src/path/pluto-cli/node/package.json +0ms
  config loadJSON /MyPath/src/path/pluto-cli/node/oclif.manifest.json +0ms
  config loadJSON /MyPath/src/path/pluto-cli/node/.oclif.manifest.json +0ms
  config:pluto-cli loading IDs from /MyPath/src/path/pluto-cli/node/src/commands +0ms
  config:pluto-cli found commands [ 'hello', 'hello:world' ] +4ms
  config loading plugins [ '@oclif/plugin-help', '@oclif/plugin-plugins' ] +12ms
  config reading core plugin /MyPath/src/path/pluto-cli/node/node_modules/@oclif/plugin-help +0ms
  config loadJSON /MyPath/src/path/pluto-cli/node/node_modules/@oclif/plugin-help/package.json +0ms
  config reading core plugin /MyPath/src/path/pluto-cli/node/node_modules/@oclif/plugin-plugins +0ms
  config loadJSON /MyPath/src/path/pluto-cli/node/node_modules/@oclif/plugin-plugins/package.json +0ms
(node:80574) SyntaxError Plugin: pluto-cli: Cannot use import statement outside a module
module: @oclif/core@2.11.3
task: toCached
plugin: pluto-cli
root: /MyPath/src/path/pluto-cli/node
See more details with DEBUG=*
(Use `node --trace-warnings ...` to show where the warning was created)
/MyPath/src/path/pluto-cli/node/src/commands/hello/index.ts:1
import {Args, Command, Flags} from '@oclif/core'
^^^^^^

SyntaxError: Cannot use import statement outside a module
    at internalCompileFunction (node:internal/vm:73:18)
    at wrapSafe (node:internal/modules/cjs/loader:1178:20)
    at Module._compile (node:internal/modules/cjs/loader:1220:27)
    at Module._extensions..js (node:internal/modules/cjs/loader:1310:10)
    at Module.load (node:internal/modules/cjs/loader:1119:32)
    at Module._load (node:internal/modules/cjs/loader:960:12)
    at Module.require (node:internal/modules/cjs/loader:1143:19)
    at require (node:internal/modules/cjs/helpers:110:18)
    at ModuleLoader.loadWithData (/MyPath/.nvm/versions/node/v18.17.0/lib/node_modules/oclif/node_modules/@oclif/core/lib/module-loader.js:80:88)
    at fetch (/MyPath/.nvm/versions/node/v18.17.0/lib/node_modules/oclif/node_modules/@oclif/core/lib/config/plugin.js:192:83)
module: @oclif/core@2.11.3
task: toCached
plugin: pluto-cli
root: /MyPath/src/path/pluto-cli/node
See more details with DEBUG=*
(node:80574) SyntaxError Plugin: pluto-cli: Cannot use import statement outside a module
module: @oclif/core@2.11.3
task: toCached
plugin: pluto-cli
root: /MyPath/src/path/pluto-cli/node
See more details with DEBUG=*
/MyPath/src/path/pluto-cli/node/src/commands/hello/world.ts:1
import {Command} from '@oclif/core'
^^^^^^

SyntaxError: Cannot use import statement outside a module
    at internalCompileFunction (node:internal/vm:73:18)
    at wrapSafe (node:internal/modules/cjs/loader:1178:20)
    at Module._compile (node:internal/modules/cjs/loader:1220:27)
    at Module._extensions..js (node:internal/modules/cjs/loader:1310:10)
    at Module.load (node:internal/modules/cjs/loader:1119:32)
    at Module._load (node:internal/modules/cjs/loader:960:12)
    at Module.require (node:internal/modules/cjs/loader:1143:19)
    at require (node:internal/modules/cjs/helpers:110:18)
    at ModuleLoader.loadWithData (/MyPath/.nvm/versions/node/v18.17.0/lib/node_modules/oclif/node_modules/@oclif/core/lib/module-loader.js:80:88)
    at fetch (/MyPath/.nvm/versions/node/v18.17.0/lib/node_modules/oclif/node_modules/@oclif/core/lib/config/plugin.js:192:83)
module: @oclif/core@2.11.3
task: toCached
plugin: pluto-cli
root: /MyPath/src/path/pluto-cli/node
See more details with DEBUG=*
  config loadJSON /MyPath/src/path/pluto-cli/node/node_modules/@oclif/plugin-help/oclif.manifest.json +0ms
  config loadJSON /MyPath/src/path/pluto-cli/node/node_modules/@oclif/plugin-plugins/oclif.manifest.json +0ms
  config:@oclif/plugin-help using manifest from /MyPath/src/path/pluto-cli/node/node_modules/@oclif/plugin-help/oclif.manifest.json +0ms
  config:@oclif/plugin-plugins using manifest from /MyPath/src/path/pluto-cli/node/node_modules/@oclif/plugin-plugins/oclif.manifest.json +0ms
  config config done +2ms
  oclif gathering workspace for pluto to /MyPath/src/path/pluto-cli/node/tmp/pluto +0ms
Error: Command failed: npm pack --unsafe-perm
npm ERR! code 1
npm ERR! path /MyPath/src/path/pluto-cli/node
npm ERR! command failed
npm ERR! command sh -c npm run build && oclif manifest && oclif readme
npm ERR! > pluto-cli@0.0.0 build
npm ERR! > shx rm -rf dist && tsc -b
npm ERR! 2023-07-31T18:29:27.158Z config reading core plugin /MyPath/src/path/pluto-cli/node/node_modules/oclif
npm ERR! 2023-07-31T18:29:27.158Z config loadJSON /MyPath/src/path/pluto-cli/node/node_modules/oclif/package.json
npm ERR! 2023-07-31T18:29:27.160Z config loadJSON /MyPath/src/path/pluto-cli/node/node_modules/oclif/oclif.manifest.json
npm ERR! 2023-07-31T18:29:27.160Z config loadJSON /MyPath/src/path/pluto-cli/node/node_modules/oclif/.oclif.manifest.json
npm ERR! 2023-07-31T18:29:27.160Z config:oclif using manifest from /MyPath/src/path/pluto-cli/node/node_modules/oclif/.oclif.manifest.json
npm ERR! 2023-07-31T18:29:27.167Z config reading user plugins pjson /MyPath/.local/share/oclif/package.json
npm ERR! 2023-07-31T18:29:27.167Z config loadJSON /MyPath/.local/share/oclif/package.json
npm ERR! 2023-07-31T18:29:27.167Z config loading plugins [
npm ERR!   '@oclif/plugin-help',
npm ERR!   '@oclif/plugin-warn-if-update-available',
npm ERR!   '@oclif/plugin-not-found'
npm ERR! ]
npm ERR! 2023-07-31T18:29:27.169Z config reading core plugin /MyPath/src/path/pluto-cli/node/node_modules/@oclif/plugin-help
npm ERR! 2023-07-31T18:29:27.169Z config loadJSON /MyPath/src/path/pluto-cli/node/node_modules/@oclif/plugin-help/package.json
npm ERR! 2023-07-31T18:29:27.169Z config reading core plugin /MyPath/src/path/pluto-cli/node/node_modules/@oclif/plugin-warn-if-update-available
npm ERR! 2023-07-31T18:29:27.169Z config loadJSON /MyPath/src/path/pluto-cli/node/node_modules/@oclif/plugin-warn-if-update-available/package.json
npm ERR! 2023-07-31T18:29:27.169Z config reading core plugin /MyPath/src/path/pluto-cli/node/node_modules/@oclif/plugin-not-found
npm ERR! 2023-07-31T18:29:27.169Z config loadJSON /MyPath/src/path/pluto-cli/node/node_modules/@oclif/plugin-not-found/package.json
npm ERR! 2023-07-31T18:29:27.169Z config loadJSON /MyPath/src/path/pluto-cli/node/node_modules/@oclif/plugin-warn-if-update-available/oclif.manifest.json
npm ERR! 2023-07-31T18:29:27.169Z config loadJSON /MyPath/src/path/pluto-cli/node/node_modules/@oclif/plugin-help/oclif.manifest.json
npm ERR! 2023-07-31T18:29:27.169Z config loadJSON /MyPath/src/path/pluto-cli/node/node_modules/@oclif/plugin-not-found/oclif.manifest.json
npm ERR! 2023-07-31T18:29:27.170Z config:@oclif/plugin-help using manifest from /MyPath/src/path/pluto-cli/node/node_modules/@oclif/plugin-help/oclif.manifest.json
npm ERR! 2023-07-31T18:29:27.170Z config:@oclif/plugin-warn-if-update-available using manifest from /MyPath/src/path/pluto-cli/node/node_modules/@oclif/plugin-warn-if-update-available/oclif.manifest.json
npm ERR! 2023-07-31T18:29:27.170Z config:@oclif/plugin-not-found using manifest from /MyPath/src/path/pluto-cli/node/node_modules/@oclif/plugin-not-found/oclif.manifest.json
npm ERR! 2023-07-31T18:29:27.170Z config config done
npm ERR! 2023-07-31T18:29:27.170Z config start init hook
npm ERR! 2023-07-31T18:29:27.291Z oclif:@oclif/plugin-warn-if-update-available:hooks:init start (require) /MyPath/src/path/pluto-cli/node/node_modules/@oclif/plugin-warn-if-update-available/lib/hooks/init/check-update.js
npm ERR! 2023-07-31T18:29:27.293Z oclif:@oclif/plugin-warn-if-update-available:hooks:init done
npm ERR! 2023-07-31T18:29:27.294Z config init hook done
npm ERR! 2023-07-31T18:29:27.294Z config runCommand manifest []
npm ERR! 2023-07-31T18:29:27.306Z config:oclif (require) /MyPath/src/path/pluto-cli/node/node_modules/oclif/lib/commands/manifest.js
npm ERR! 2023-07-31T18:29:27.306Z config start prerun hook
npm ERR! 2023-07-31T18:29:27.306Z config prerun hook done
npm ERR! 2023-07-31T18:29:27.306Z oclif:manifest init version: @oclif/core@2.11.1 argv: []
npm ERR! 2023-07-31T18:29:27.309Z config reading core plugin /MyPath/src/path/pluto-cli/node
npm ERR! 2023-07-31T18:29:27.309Z config loadJSON /MyPath/src/path/pluto-cli/node/package.json
npm ERR! 2023-07-31T18:29:27.312Z config:pluto-cli loading IDs from /MyPath/src/path/pluto-cli/node/src/commands
npm ERR! 2023-07-31T18:29:27.316Z config:pluto-cli found commands [ 'hello', 'hello:world' ]
npm ERR! /MyPath/src/path/pluto-cli/node/src/commands/hello/index.ts:1
npm ERR! import {Args, Command, Flags} from '@oclif/core'
npm ERR! ^^^^^^
npm ERR! 
npm ERR! SyntaxError: Cannot use import statement outside a module
npm ERR!     at internalCompileFunction (node:internal/vm:73:18)
npm ERR!     at wrapSafe (node:internal/modules/cjs/loader:1178:20)
npm ERR!     at Module._compile (node:internal/modules/cjs/loader:1220:27)
npm ERR!     at Module._extensions..js (node:internal/modules/cjs/loader:1310:10)
npm ERR!     at Module.load (node:internal/modules/cjs/loader:1119:32)
npm ERR!     at Module._load (node:internal/modules/cjs/loader:960:12)
npm ERR!     at Module.require (node:internal/modules/cjs/loader:1143:19)
npm ERR!     at require (node:internal/modules/cjs/helpers:110:18)
npm ERR!     at ModuleLoader.loadWithData (/MyPath/src/path/pluto-cli/node/node_modules/@oclif/core/lib/module-loader.js:80:88)
npm ERR!     at fetch (/MyPath/src/path/pluto-cli/node/node_modules/@oclif/core/lib/config/plugin.js:192:83)

npm ERR! A complete log of this run can be found in: /MyPath/.npm/_logs/2023-07-31T18_29_25_421Z-debug-0.log

    at ChildProcess.exithandler (node:child_process:419:12)
    at ChildProcess.emit (node:events:514:28)
    at maybeClose (node:internal/child_process:1091:16)
    at ChildProcess._handle.onexit (node:internal/child_process:302:5)
andrewjenkins@MacBook-Pro node % 
```

**Comment by mdonnalley** [comment]
@drewjenkins It looks like it's failing to compile the project on both `generate` and `pack tarballs`

Once you cd into the directory, are you able to successfully run `npm run build`?

**Comment by drewjenkins** [comment]
Yep the build works like a champ and the files are generated in dist

![image](https://github.com/oclif/oclif/assets/5521743/d1f385d1-af6e-419f-988d-10ce43c584a9)

**Comment by drewjenkins** [comment]
Had a co-worker do the same above steps, he can't repro either.  used same oclif and node versions.  I tried a few different things.  Moving the project to a different folder (outside of my work folder, in case something funky was going on). Trying yarn instead of NPM. Trying ESM instead of CommonJS.  Nothing working.  It must be some configuration locally with my computer though.  At this point I'm considering reinstalling MacOS fresh :| I think we can probably close this out since it seems to just be a me issue

**Comment by drewjenkins** [comment]
Ok, so after formatting and reinstalling MacOS completely fresh, and reinstalling all apps fresh, and using my old .zshrc I was still hitting the issue.  Banging my head against the wall I started trying to eliminate any possible environmental factor, so I started doing things as vanilla as possible.

Afte much trial and error and commenting things out, I ran the problem down to this line in my `.zshrc`

`export NODE_ENV=development`

I don't remember exactly why I had it set like that... it was some project I was working on suggested to do it.  I knew it was a bad idea at the time because I knew it would come back to bite me in the butt, and here we are.  Commented that line out and everything is good to go now.

Closing the issue out
